src/legacy/core_plugins/kibana/public/management/sections/objects/lib/in_app_url.ts

```diff
@@ -17,22 +17,24 @@
  * under the License.
  */
 
-export function canViewInApp(uiCapabilities, type) {
+import { Capabilities } from 'src/core/public';
+
+export function canViewInApp(uiCapabilities: Capabilities, type: string): boolean {
   switch (type) {
     case 'search':
     case 'searches':
-      return uiCapabilities.discover.show;
+      return uiCapabilities.discover.show as boolean;
```

Comment:
I'm surprised that we don't allow to specify the type of `Capabilities` via a generic interface

Reply:
Yes, this is something that was 'migrated' a little too fast to TS imho. Adding a genetic type on `Capabilities` would probably allow to remove quite a lot of force-cast in our TS plugins code.

However something like `Capabilities<T> extends T` is not permitted, so I'm not sure how we would allow to add arbitrary root-level properties on `Capabilities` to replace 

```ts
  /** Custom capabilities, registered by plugins. */
  [key: string]: Record<string, boolean | Record<string, boolean>>;
```